### PR TITLE
FIDEFE-4640 - Add MPL license and how to edit file header

### PR DESCRIPTION
**Pull request by TGiles**
This patch adds the MPL license header to the generated files and also a comment on how to change the generated output. I'm open to changes on the wording as well as the command that I threw in there (since we don't have a `./mach build-design-tokens` command or anything yet)

### CODE REVIEW

File: toolkit/themes/shared/design-system/build/css/tokens-shared.css [hunk 1]
```diff
@@ -1,3 +1,12 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ * Do not edit this file directly, instead modify design-tokens.json
+ * and run `npm run build` to see your changes.
```
Comment (hannajones) [inline toolkit/themes/shared/design-system/build/css/tokens-shared.css:9]: nit: imo it would be nice to separate this from the license comment block - right now they kinda look related. This should be more doable now that we have a formatter.

File: toolkit/themes/shared/design-system/config.js [hunk 2]
```diff
@@ -45,7 +53,11 @@ module.exports = {
           format: "css/variables",
           options: {
             outputReferences: true,
-            showFileHeader: false,
+            fileHeader: () => [
```
Comment (hannajones) [inline toolkit/themes/shared/design-system/config.js]: I think it'll work better for us long term to define the custom header in such a way that we can reference it in multiple formats, like how they do it in this example: https://github.com/amzn/style-dictionary/blob/main/examples/advanced/custom-file-header/build.js

If that seems out of scope here though we can just change it when we add the different formats for brand, platform, etc.

**Review by hannajones — APPROVED** [review]
Works for me, thanks! A couple of optional suggestions, feel free to take or leave them.

File: toolkit/themes/shared/design-system/config.js [hunk 3]
```diff
@@ -24,6 +49,7 @@ const MEDIA_QUERY_PROPERTY_MAP = {
  */
 function hcmFormatter(args) {
   return (
+    customFileHeader() +
```
Comment (hannajones) [inline toolkit/themes/shared/design-system/config.js:52]: nice! this seems way better than having to reference the `fileHeader` in three different places when we have multiple formats 😄

**Review by hannajones — APPROVED** [review]
LGTM!

**Review by mstriemer — APPROVED** [review]
Looks good!
